ncm-filesystems/src/main/perl/filesystems.pm

```diff
@@ -1,116 +1,148 @@
+            $self->verbose ("Scheduling for removal filesystem $mount") if $remove;
+    	}
+    }
+
+    foreach my $outdated (@rm) {
+        if ($remove) {
+            my $fsrm = NCM::Filesystem->new_from_fstab ($outdated, $config, log => $self);
+            $self->info("Removing filesystem for $fsrm->{mountpoint}");
+            $fsrm->remove_if_needed==0 or return 0;
+        }
+        $self->info("Removing line $outdated");
+        $fstab->replace_lines (qr{$outdated}, qr{^$}, "");
+
+    }
+    return 1;
 }
 
+
+sub create_blockdevices
+{
+    my ($self, $config, $fs) = @_;
+    # Partitions must be created first
+    my $el = $config->getElement ("/system/blockdevices/partitions");
+    my @part = ();
+    $self->info ("Checking whether partitions need to be created");
+
+    while ($el && $el->hasNextElement) {
+        my $el2 = $el->getNextElement;
+        push (@part, NCM::Partition->new ($el2->getPath->toString, $config, log => $self));
+    }
+    foreach my $partition (sort partition_compare @part) {
+        if ($partition->create != 0) {
+            $self->error("Couldn't create partition: " . $partition->devpath);
+            return 0;
+        }
+    }
+    foreach my $filesystem (@$fs) {
+        if ($filesystem->create_if_needed != 0) {
+            $self->error("Failed to create filesystem:  $filesystem->{mountpoint}");
+            return 0;
+        }
+    }
+    return 1;
+};
+
+
 sub Configure
 {
-	my ($self, $config) = @_;
-
-	my @fs = ();
-
-	if ($CAF::Object::NoAction) {
-		$self->warn ("--noaction not supported. Leaving.");
-		return 1;
-	}
-
-	my $el = $config->getElement ("/system/filesystems");
-	while ($el->hasNextElement) {
-		my $el2 = $el->getNextElement;
-		push (@fs, NCM::Filesystem->new ($el2->getPath->toString, $config, log => $self));
-	}
-	my %protected = protected_hash ($config);
-	$self->free_space ($config, \%protected, @fs)==0 or return 0;
-	# Partitions must be created first, see bug #26137
-	$el = $config->getElement ("/system/blockdevices/partitions");
-	my @part = ();
-	$self->info ("Checking whether partitions need to be created");
-
-	while ($el && $el->hasNextElement) {
-		my $el2 = $el->getNextElement;
-		push (@part, NCM::Partition->new ($el2->getPath->toString, $config, log => $self));
-	}
-	foreach (sort partition_compare @part) {
-		if ($_->create != 0) {
-			throw_error ("Couldn't create partition: " . $_->devpath);
-			return 0;
-		}
-	}
-	foreach (@fs) {
-		if ($_->create_if_needed != 0) {
-			throw_error ("Failed to create filesystem:  $_->{mountpoint}");
-			return 0;
-		}
-
-		if ($_->format_if_needed (%protected) != 0) {
-			$self->warn ("Failed to format filesystem: ".
-				      $_->{mountpoint});
-		}
-	}
-	return 1;
+    my ($self, $config) = @_;
+
+    my $tree = $config->getTree($self->prefix());
+    my $fstab_tree = $config->getTree($FSTAB_CMP_PATH);
+
+    my $manage = $tree->{manage_blockdevs};
+    $self->info('Managing blockdevices: ', ($manage) ? 'yes' : 'no');
```

Comment:
Can you add a note about the manage_blockdevs attribute to the .POD since it's not referenced. Also I'm not sure I know why this setting is useful: if you want this behaviour, why not set this component to active = false and use ncm-fstab?